File: viral-cast-ai-frontend/src/lib/types/regions.ts
export interface PaginationInfo {
	total: number;
	limit: number;
	offset: number;
	has_next: boolean;
}

export interface RegionsListResponse<T> {
	success: boolean;
	message: string;
	data: T[];
	pagination: PaginationInfo;
}

export interface SingleRegionResponse<T> {
	success: boolean;
	message: string;
	data: T;
}

export interface Province {
	uuid: string;
	code: string;
	name: string;
}

export interface Regency {
	uuid: string;
	code: string;
	name: string;
	province_uuid: string;
	province_name: string;
	// Optional raw code from backend when available
	province_code?: string;
}

export interface District {
	uuid: string;
	code: string;
	name: string;
	// The documentation example uses "district_uuid" pointing to regency; assume it's a typo and use regency_uuid
	regency_uuid?: string;
	regency_name?: string;
	province_uuid: string;
	province_name: string;
	// Optional raw codes from backend when available
	regency_code?: string;
	province_code?: string;
}

export interface Village {
	uuid: string;
	code: string;
	name: string;
	district_uuid: string;
	district_name: string;
	regency_uuid: string;
	regency_name: string;
	province_uuid: string;
	province_name: string;
	// Optional raw codes from backend when available
	district_code?: string;
	regency_code?: string;
	province_code?: string;
}

File: viral-cast-ai-backend/src/models/regions.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Province {
    pub uuid: Uuid,
    pub code: String,
    pub name: String,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Regency {
    pub uuid: Uuid,
    pub code: String,
    pub name: String,
    pub province_uuid: Uuid,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct District {
    pub uuid: Uuid,
    pub code: String,
    pub name: String,
    pub regency_uuid: Uuid,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Village {
    pub uuid: Uuid,
    pub code: String,
    pub name: String,
    pub district_uuid: Uuid,
}

// View models with hierarchical information
#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct ProvinceView {
    pub code: String,
    pub name: String,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct RegencyView {
    pub code: String,
    pub name: String,
    pub province_code: String,
    pub province_name: String,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct DistrictView {
    pub code: String,
    pub name: String,
    pub regency_code: String,
    pub regency_name: String,
    pub province_code: String,
    pub province_name: String,
}

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct VillageView {
    pub code: String,
    pub name: String,
    pub district_code: String,
    pub district_name: String,
    pub regency_code: String,
    pub regency_name: String,
    pub province_code: String,
    pub province_name: String,
}
